### ISSUE TITLE
Officially support KDE5

### ISSUE THREAD

**Issue by wolftune** [KXStudio/Repository#58]
Given that I believe KXStudio repos work with Ubuntu 16.04 already, the only other factor is to work with KDE5. We need to test this with Kubuntu or Neon or other options. Any issues need to be worked out, and then KXStudio can formally support KDE5 etc (I mean to include all the relevant frameworks, plasma, qt5, the whole list of all the things that are updated in the latest KDE).

**Comment by wolftune** [comment]
Issues discovered when trying KDE Neon and KXStudio repos:

* KXStudio repos are still installing ardour4 package instead of ardour
* Various broken/held/unresolvable errors when trying to install kxstudio-meta-all
    * Installed kxstudio-meta-audio, graphics, wine, video, restricted-extras successfully
    * had errors trying to install meta-audio-plugins
    * a specific error case for eq10: http://kxstudio.linuxaudio.org/paste/r09W6
    * other packages like guitarix have the same errors
* Cadence tools (probably because using qt4) fail to grey out buttons like "stop" when they are inactive
* will edit this comment as more is clarified…

**Comment by falkTX** [comment]
1. We can't force the installation of ardour right now because of a bug in apt in latest debian testing/sid.
(it fails to download the deb file from github)
I don't want to have non-installable meta-packages.

2. Regarding meta-all, etc. You're missing the gcc5 extra repos.
The dependencies for eq10q are these in the gcc5 variant:
`Depends: libatkmm-1.6-1v5 (>= 2.22.1), libc6 (>= 2.15), libcairomm-1.0-1v5 (>= 1.6.4), libgcc1 (>= 1:4.1.1), libglibmm-2.4-1v5 (>= 2.44.0), libgtkmm-2.4-1v5 (>= 1:2.24.0), libpangomm-1.4-1v5 (>= 2.36.0), libsigc++-2.0-0v5 (>= 2.2.0), libstdc++6 (>= 5.2)`

3. Cadence not having greyed out buttons is a bug in Qt4? I think. Qt4 apps are broken under KDE5.

4. Regarding firefox, try using plasmazilla PPA:
https://launchpad.net/~plasmazilla/+archive/ubuntu/releases
The 14.04 ISOs use that too.

5. Has the redshift plasmoid been ported to KDE5?

6. Thanks for apt-xapian-index hint, I guess we need to add as dependency to desktop-base.

7. KMixer requires you to disable the native plasma mixer first.
Also make sure this is in your ~/.bashrc or ~/.bash_aliases:
`export KMIX_PULSEAUDIO_DISABLE=1`
Then relogin into KDE.

**Comment by wolftune** [comment]
Sorry to confuse you but I kept updating the report as I went.

1. Thanks for the clarification
2. I got it all working already, but thanks
3. Right, I mean it's unfortunate to not have greyed buttons because of qt4 / KDE5, would be best to port to qt5
4. I'll look into it. Mostly it's fine, it's mainly just the file save/open dialog that stinks
5. There's a NEW redshift plasmoid for KDE5 called "Redshift Control" and it's available in the plasma "add widgets" functionality

\7. OH!! kmix is a different package! Got it! It works now, just some setup! Yay!

Okay, basically this whole system is usable! I've had few issues overall. I think I can do this!

I would like better clarity about the correct order by which a new install adds the various kxstudio meta and settings packages etc., that wasn't clear to me. I think I get it now though.

What other items should I test?

**Comment by falkTX** [comment]
hmm, see if the kxstudio-apply-root still works for KDE5.

I have 16.04 at home now, so for regular things I can test too.
I've enabled kubuntu backports though, I need that multi-monitor support...

Does breeze-gtk3 follow the system colors? I believe it doesn't...
We need to see what to do regarding gtk3 themes.

**Comment by falkTX** [comment]
btw, I created a ppa for testing KDE5 stuff in 16.04.
ppa:kxstudio-debian/kde5-testing

There's nothing in there for now, but if we need anything I'll upload it there.
(requires kubuntu backports)

**Comment by falkTX** [comment]
Oh, 1 thing we'll need is a kxstudio-desktop-kde5 meta-package.

I uploaded an update to kxstudio-artwork to the new ppa.
It includes the "old" kxstudio's qtcurve and color schemes for KDE5.

**Comment by wolftune** [comment]
Although it's nice for the artwork to be updated, it's not as big a deal. I don't really need the artwork to have a usable system. Incidentally, Neon has no qtcurve at all by default. I didn't check if qtcurve itself is updated to qt5 yet.

**Comment by falkTX** [comment]
The artwork was just for me :P

There is qtcurve-qt5 already, but the issue of qt4 disabled widgets not rendering properly under KDE5 still exists for qtcurve-qt4.

**Comment by wolftune** [comment]
dependency problem install musescore! It looks like aptitude was able to solve it by using the Ubuntu package instead of the KXStudio package. I'm still stuck at older version though, see #35

**Comment by falkTX** [comment]
what's the problem with musescore? installs fine here on 16.04.

**Comment by wolftune** [comment]
Musescore now works fine, but when I first tried to install with apt, it had conflicts for some reason, couldn't get it installed until I used aptitude to better seek out some alternate dependency approach…

**Comment by wolftune** [comment]
note: the plasma firefox stuff doesn't work at all, it downgraded firefox and then wouldn't run at all, no terminal message. I couldn't find out enough about what it does to determine that it even makes sense. I had to to a roll-back to get it working. It's possible I could have logged out and back in and other stuff and got it working, but I don't like that FF 48 via the plasma PPA was marked as newer than FF 49, that's an awkward hack. I think the Neon folks would use any appropriate firefox adjustments they are comfortable with.

**Comment by falkTX** [comment]
The firefox stuff does work for me..

The reason why it downgrades is because firefox needs to be manually patched, which often requires additional work. Same for thunderbird, although not so often.

**Comment by wolftune** [comment]
Is there any where that explains what the firefox patching achieves?

**Comment by falkTX** [comment]
The patch adds support for KDE file dialogs (for open/save) and KDE application browser when selecting which app to open which file.
Basically it adds KDE integration.
Firefox devs said they won't merge it because it links to KDE libs, which they do not want.

**Comment by wolftune** [comment]
Ok, I'll try it again to get it working. My point was that the *Neon* folks should be including the patch, not core Firefox devs per se.

**Comment by falkTX** [comment]
It's the other way around.
Firefox devs should merge the patch so that distros don't have to keep patching it.

**Comment by wolftune** [comment]
Right, in terms of how things *ought* to be. I just mean that I'd realistically *expect* Neon to have it versus Firefox devs accepting it.

**Comment by wolftune** [comment]
Okay, I got it working this time. Firefox then froze on me, but we'll see…

**Comment by wolftune** [comment]
I'm getting that obnoxious popup about KDE losing some audio devices and should I accept forgetting them or not blah blah. I believe this was patched in KDE in the regular KXStudio release. I would like to get an updated patch that kills that dialog.

**Comment by wolftune** [comment]
I got Cadence to show the xruns number in a launcher when set to always show a launcher icon. But when I click that icon, it now opens a second icon in the task manager that goes with the actual window and that icon has no indicator. The python window isn't being recognized as being the program the launcher applies to.

**Comment by wolftune** [comment]
getting errors with alsa bridge reconnecting when computer wakes from suspend:

```
Tue Oct  4 09:08:30 2016: ERROR: JackEngine::XRun: client = alsa2jack was not finished, state = Triggered
Tue Oct  4 09:08:30 2016: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
```

**Comment by wolftune** [comment]
added little confused icon on items that need attention still. Mostly, things are okay. Just little details. I have a usable system!

**Comment by falkTX** [comment]
I think we need to make a list of items for proper KDE5 support.
This bug report can serve as a meta-bug for general KDE5 talk, while every little detail should have its own bug. Otherwise it will get very hard to track what needs to be done.
I'll create a few now.

**Comment by falkTX** [comment]
I think the not-inclusion of apt-xapian-index is intentional, afaik it can randomly create high disk usage by indexing package data.
That's a very bad thing to happen if, for example, you're recording audio.

**Comment by ciacnorris** [comment]
Another minor but not-that-unimportant thing could be the CPU frequency governor applet: I have never found one for Plasma5 that has the same features of the one in the actual KXStudio distro.
BTW, should you need any help in testing (especially on Linux Mint KDE 18.1, but also on a fresh install of something else) let me know.

**Comment by ciacnorris** [comment]
News about that here:
https://forum.manjaro.org/t/applet-to-change-cpu-governor-frequency-in-kde-edition/
(Yes, I'm quite obsessed with it :-) )

**Comment by falkTX** [comment]
I am closing this issue since it we got meta-neon now.
It is not KDE5 directly, but it is the desktop the next ISO will use.
Other issues need to be reported separately
